managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -1685,6 +1686,20 @@ void asyncReadEntries(OpReadEntry opReadEntry) {
         return getLedgerHandle(ledgerId).thenApply(rh -> rh.getLedgerMetadata().toSafeString());
     }
 
+    @Override
+    public CompletableFuture<LedgerInfo> getClosedLedgerInfo(long ledgerId) throws ManagedLedgerException {
+        final LedgerInfo ledgerInfo = ledgers.get(ledgerId);
+        if (ledgerInfo == null) {
+            throw new ManagedLedgerException(
+                    Strings.lenientFormat("ledger with id %s not found", ledgerId));
+        } else if (ledgerInfo.getSize() == 0 || ledgerInfo.getEntries() == 0) {
```

Comment:
> What happens if the ledger is closed with zero entries?

It will be deleted from metadata, see https://github.com/apache/pulsar/blob/d3a6a8c2744733d00e6dfa30c0cac2f06eb1386a/managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java#L1535-L1547